### PR TITLE
OS-X / Linux automated installer, make-compatible; support dot in the function names.

### PR DESCRIPTION
**Pull request by kigster**
* On my system default awk was throwing errors.
 * Installing `gawk` solved the issue
 * Adding extensible setup script that can be easily extended to install on other OSes. Linux and OS-X should be supported out the box.
 * My function namespaces are separated by a dot, not by colon — so adding dot to the regex.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2017 Stanislav Seletskiy & Egor Kovetskiy
+Copyright © 2017-2020 Stanislav Seletskiy & Egor Kovetskiy
```
Comment (tomjn) [inline LICENSE:3]: This could be interpreted that copyright has been relinquished from 2020 onwards

Comment (kigster) [reply LICENSE:3]: I don't believe so, as long as it's the current and not past year it remains with the authors. 

I mean it's a total obsessive compulsive nit pick, to be fixing other folks copyrights year ranges, so perhaps I should get it checked out 😃

**Comment by tomjn** [comment]
I think the dot in function names change can be separated out into its own PR to make this more atomic and both easier to test

**Comment by seletskiy** [comment]
@kigster Thanks for the contribution.

I think that OSX installation is out of the scope for the current project and the install script should be moved to the brew package repository.

Adding dot to the function names is, however, nice.

**Pull request closed without merging**